docs/sdk/web3-react/guides/01-connect-wallet.md

```diff
@@ -1,4 +1,165 @@
 ---
 id: connect-wallet
 title: Connecting to Wallets
----     
\ No newline at end of file
+---     
+
+## Introduction
+
+This guide will cover how to connect wallets with `web3-react`. It is based on the [web3-react example](https://github.com/Uniswap/examples/tree/main/web3-react), found in the Uniswap code examples [repository](https://github.com/Uniswap/examples). To run this example, check out the examples's [README](https://github.com/Uniswap/examples/blob/main/web3-react/README.md) and follow the setup instructions.
+
+
+In this example we will walk through setting up `web3-react` and connecting the most popular browser injected connector, *MetaMask*, using the web3-react [MetaMask package](https://www.npmjs.com/package/@web3-react/metamask).
+
+
+The input parameters to this guide are the chains that we want our app to be able to connect to and their RPC URLs.
+
+The guide will **cover**:
+
+1. Creating a `Web3ReactProvider`
+2. Building an `InjectedConnector`
+3. Connecting and  disconnecting the application to the connector
+
+At the end of the guide, we should be able to connect and disconnect the application to a MetaMask injected connector.
+
+For this guide, the following `web3-react` packages are used:
+
+- [`@web3-react/core`](https://www.npmjs.com/package/@web3-react/core)
+- [`@web3-react/types`](https://www.npmjs.com/package/@web3-react/types)
+- [`@web3-react/metamask`](https://www.npmjs.com/package/@web3-react/metamask)
+
+The core code of this guide can be found in [Web3Provider](https://github.com/Uniswap/examples) and [Injected Connector](https://github.com/Uniswap/examples).
+
+## Creating a `Web3ReactProvider`
+
+To be able to interact with the methods that `web3-react` offers, we first need to setup a `Web3ReactProvider` and wrap our application in it. `web3-react` uses [React Context](https://reactjs.org/docs/context.html) to allow us to use the exposed hooks without additional configuration. 
+
+To start, we create a React component called `Web3ContextProvider` in order to wrap the logic of configuring the `Web3ReactProvider`. In this component, we first import  `Web3ReactProvider` from the web3-react [core package](https://www.npmjs.com/package/@web3-react/core).
+
+The component receives just one parameter which is the children to which it will be providing the React Context:
```

Comment:
refer to this as a prop to be more idiomatic to React:

    The component receives just one prop - the `children` which will be the Context's consumers: